openapi/find-my-trn-api.yml

```
@@ -0,0 +1,291 @@
+              schema:
+                type: string
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/trndetails'
+
+  /trn-details-match:
+    get:
+      summary: 'teacher endpoint: returns trn details given an email'
+      parameters:
+        - name: emailaddress
+          in: query
+          description: email address
+          required: false
+          schema:
+            type: string
+        
+        - name: firstname
+          in: query
+          description: first name of person
+          required: false
+          schema:
+            type: string
+        
+        - name: middlename
+          in: query
+          description: middle name of person
+          required: false
+          schema:
+            type: string
+
+        - name: lastname
+          in: query
+          description: last name of person
+          required: false
+          schema:
+            type: string
+        
+        - name: previousfirstname
+          in: query
+          description: pervious first name of person
+          required: false
+          schema:
+            type: string
+        
+        - name: previouslastname
+          in: query
+          description: previous last name of person
+          required: false
+          schema:
+            type: string
+        
+        - name: dateofbirth
+          in: query
+          description: dob of person
+          required: false
+          schema:
+            type: string
+
+        - name: nino
+          in: query
+          description: nino of person
+          required: false
+          schema:
+            type: string         
+        
+        - name: ittprovidername
+          in: query
+          description:  name of teacher training provider
+          required: false
+          schema:
+            type: string
+
+        - name: ittproviderukprn
+          in: query
+          description: ukprn of itt provider
+          required: false
+          schema:
+            type: string
+
+      responses:
+        '200':
+          description: 'Ok'
+          headers:
+            x-next:
+              description: A link to the next page of responses
+              schema:
+                type: string
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/trndetails'
+
+  /itt-establishments:
+      get:
+        summary: 'reference list of: itt establishment details'
+              
+        responses:
+          '200':
+            description: 'Ok'
+            headers:
+              x-next:
+                description: A link to the next page of responses
+                schema:
+                  type: string
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/ittestablishments'
+
+  /unlock-trn-trecord:
+    get:
```

Should this be a `put`?

```suggestion
    put:
```

---

Or maybe POST?

---

yep it should.

---

I have made it a PUT. As I think that best describes the idempotency of unlocking and because we know the ID of the row we want to update.